config/confighttp/compression.go

```diff
@@ -82,19 +82,21 @@ func (r *compressRoundTripper) RoundTrip(req *http.Request) (*http.Response, err
 	if writerErr != nil {
 		return nil, writerErr
 	}
-	_, copyErr := io.Copy(compressWriter, req.Body)
-	closeErr := req.Body.Close()
+	if req.Body != nil {
+		_, copyErr := io.Copy(compressWriter, req.Body)
+		closeErr := req.Body.Close()
 
-	if err := compressWriter.Close(); err != nil {
-		return nil, err
-	}
+		if copyErr != nil {
```

Comment:
How hard would it be to test this condition?

Reply:
I add unit test `TestHTTPContentCompressionCopyError`.

Reply:
Yes you are right, I've modified it and the coverage is not a problem any more. @jpkrohling